presto-verifier/src/main/java/com/facebook/presto/verifier/framework/DdlVerification.java

```diff
@@ -71,15 +81,49 @@ protected DdlMatchResult verify(
             ChecksumQueryContext controlChecksumQueryContext,
             ChecksumQueryContext testChecksumQueryContext)
     {
-        Statement controlChecksumQuery = getChecksumQuery(control);
-        Statement testChecksumQuery = getChecksumQuery(test);
+        String controlChecksum = null;
+
+        if (!skipControl || saveSnapshot) {
+            Statement controlChecksumQuery = getChecksumQuery(control);
+            controlChecksumQueryContext.setChecksumQuery(formatSql(controlChecksumQuery));
+
+            controlChecksum = getOnlyElement(callAndConsume(
+                    () -> getHelperAction().execute(controlChecksumQuery, CONTROL_CHECKSUM, checksumConverter),
+                    stats -> stats.getQueryStats().map(QueryStats::getQueryId).ifPresent(controlChecksumQueryContext::setChecksumQueryId)).getResults());
+
+            if (saveSnapshot) {
+                try {
+                    ObjectMapper objectMapper = new ObjectMapper();
+                    String snapshot = objectMapper.writeValueAsString(controlChecksum);
+                    snapshotQueryConsumer.accept(new SnapshotQuery(getSourceQuery().getSuite(), getSourceQuery().getName(), isExplain, snapshot));
+                }
+                catch (JsonProcessingException exception) {
+                    throw new RuntimeException(exception);
+                }
+
+                return new DdlMatchResult(MATCH, Optional.empty(), "", "");
+            }
+        }
+        else if (QUERY_BANK_MODE.equals(runningMode)) {
+            ObjectMapper objectMapper = new ObjectMapper();
+            String key = format(VERIFIER_SNAPSHOT_KEY_PATTERN, getSourceQuery().getSuite(), getSourceQuery().getName(), isExplain);
+            SnapshotQuery snapshotQuery = snapshotQueries.get(key);
+            if (snapshotQuery != null) {
+                try {
+                    String snapshotJson = snapshotQuery.getSnapshot();
+                    controlChecksum = objectMapper.readValue(snapshotJson, String.class);
```

Comment:
same as above. Do we need an objectmapper to read the raw string back.?

Reply:
Changed. Thanks

Reply:
@ajaygeorge Same as above, restored the old code.